Source/Details/ASPageTable.m

```diff
@@ -148,4 +148,17 @@ - (void)removeObjectForPage:(ASPageCoordinate)page
   [self removeObjectForKey:key];
 }
 
+- (NSMapTable *)deepCopy
+{
+  NSMapTable *result = [[NSMapTable alloc] initWithKeyPointerFunctions:self.keyPointerFunctions
+                                                 valuePointerFunctions:self.valuePointerFunctions
+                                                              capacity:self.count];
+
+  for (id key in self) {
+    [result setObject:[[self objectForKey:key] copy] forKey:key];
+  }
```

Comment:
Two performance questions here that deserve some profiling or thought:

- `initWithKeyOptions:valueOptions:` is optimized compared to `initWithKeyPointerFunctions:valuePointerFunctions:`. I don't know whether the resulting storage also performs differently.
- `self.keyPointerFunctions` and `self.valuePointerFunctions` both create new `NSPointerFunctions` objects.

Now that measurement and layoutAttributes are totally disconnected, not sure if we even need this. Right?